awswrangler/sagemaker.py

```diff
@@ -22,34 +22,68 @@ def _parse_path(path):
         parts = path2.partition("/")
         return parts[0], parts[2]
 
-    def get_job_outputs(self, job_name: str = None, path: str = None) -> Any:
+    def get_job_outputs(self, job_name: str = None, path: str = None) -> Dict[str, Any]:
+        """
+        Extract and deserialize all Sagemaker's outputs (everything inside model.tar.gz)
+
+        :param job_name: Sagemaker's job name
+        :param path: S3 path (model.tar.gz path)
+        :return: A Dictionary with all filenames (key) and all objects (values)
+        """
 
         if path and job_name:
-            raise InvalidParameters("Specify either path, job_arn or job_name")
+            raise InvalidParameters("Specify either path or job_name")
 
         if job_name:
             path = self._client_sagemaker.describe_training_job(
                 TrainingJobName=job_name)["ModelArtifacts"]["S3ModelArtifacts"]
 
-        if not self._session.s3.does_object_exists(path):
-            return None
+        if path is not None:
+            if path.split("/")[-1] != "model.tar.gz":
+                path = f"{path}/model.tar.gz"
 
-        bucket, key = SageMaker._parse_path(path)
-        if key.split("/")[-1] != "model.tar.gz":
-            key = f"{key}/model.tar.gz"
+        if self._session.s3.does_object_exists(path) is False:
+            raise InvalidSagemakerOutput(f"Path does not exists ({path})")
 
+        bucket: str
+        key: str
+        bucket, key = SageMaker._parse_path(path)
         body = self._client_s3.get_object(Bucket=bucket, Key=key)["Body"].read()
         body = tarfile.io.BytesIO(body)  # type: ignore
         tar = tarfile.open(fileobj=body)
 
-        results = []
-        for member in tar.getmembers():
+        members = tar.getmembers()
+        if len(members) < 1:
+            raise InvalidSagemakerOutput(f"No artifacts found in {path}")
+
+        results: Dict[str, Any] = {}
+        for member in members:
+            logger.debug(f"member: {member.name}")
             f = tar.extractfile(member)
-            file_type = member.name.split(".")[-1]
+            file_type: str = member.name.split(".")[-1]
 
             if (file_type == "pkl") and (f is not None):
                 f = pickle.load(f)
 
-            results.append(f)
+            results[member.name] = f
 
         return results
+
+    def get_model(self, job_name: str = None, path: str = None, model_name: str = None) -> Any:
+        """
+        Extract and deserialize a Sagemaker's output model (.tat.gz)
+
+        :param job_name: Sagemaker's job name
+        :param path: S3 path (model.tar.gz path)
+        :param model_name: model name (e.g: )
+        :return:
```

Comment:
Missing the return information. Could be something like:
```
:return: The deserialized model, in which the type will depend on the algorithm used
```

Reply:
https://github.com/awslabs/aws-data-wrangler/pull/96